src/controllers/FileSink.ts

```diff
@@ -0,0 +1,101 @@
+import { LogEventLevel } from 'structured-log';
+import { LogEvent } from 'structured-log/logEvent';
+import { Sink } from 'structured-log/sink';
+import fs = require('fs');
+
+export class FileSink implements Sink {
+  constructor() {
+    // Create Logs/ folder if it doesn't exist
+    if (!fs.existsSync('Logs/')) {
+      fs.mkdirSync('Logs/');
+    }
+  }
+
+  /**
+   * Save 'events' to file
+   * @param events Array of events to be saved to file
+   */
+  public emit(events: LogEvent[]) {
+    for (let i = 0; i < events.length; ++i) {
+      const e = events[i];
+      
+      // Call WriteToFile based on event type
+      switch (e.level) {
+        case LogEventLevel.fatal:
+          this.WriteToFile('Fatal', e);
+          break;
+
+        case LogEventLevel.error:
+          this.WriteToFile('Error', e);
+          break;
+
+        case LogEventLevel.warning:
+          this.WriteToFile('Warning', e);
+          break;
+
+        case LogEventLevel.information:
+          this.WriteToFile('Information', e);
+          break;
+          
+        case LogEventLevel.debug:
+          this.WriteToFile('Debug', e);
+          break;
+          
+        case LogEventLevel.verbose:
+          this.WriteToFile('Verbose', e);
+          break;
+
+        default: 
+          this.WriteToFile('Log', e);
+          break;
+      }
+    }
+  }
+
+  public flush() {
+    // TODO: call flush and run appendFile here instead of every time an event is recieved
+
+    // Return resolved promise
+    return Promise.resolve();
+  }
+
+  private WriteToFile(prefix: string, e: LogEvent) {
+    // output = time [prefix] | message to write to file
+    const output = `${GetTime()} [${prefix}] | ${e.messageTemplate.render(e.properties)}`;
+
+    // filename = Logs/<current-date>.txt
+    const filename: string = 'Logs/' + GetDate() + '.txt';
+    // Write lines to file
+    fs.appendFileSync(filename, '\n' + output);
+  }
+}
+
+/**
+ * Get the current date for logfile name
+ */
+function GetDate() {
```

Comment:
In JS, the convention we use is to name regular functions with lowerCaseCamelCase. This is because UpperCaseCamelCase is used for [constructor functions](https://www.w3schools.com/js/js_object_constructors.asp), and accidentally calling a constructor function with the `new` keyword can have unintended side-effects.

Also, in this project we use the convention of assigning functions to variables like this: `const getDate = function() { `